### PR TITLE
READY: Added NodeJS/JS examples to dev/using/2i

### PR DESCRIPTION
**Pull request by lukebakken**

### CODE REVIEW

**Comment by lukebakken** [comment]
PR #1618 is inadvertently part of this. Just look at the changes to `2i.md` here.

**Comment by TheOnlyRew** [comment]
Should we close #1618 then, and collapse the PRs into one? Seems like the easier choice than rewriting the history for this branch.

File: source/languages/en/riak/dev/using/2i.md [hunk 1]
```diff
@@ -596,6 +722,29 @@ riak.RiakError: "Riak 2i fields must end with either '_bin' or '_int'."
...
+    .withBucket('people')
+    .withIndexName('field2_foo')
+    .withIndexKey('jsmith123')
+    .withCallback(query_cb)
+    .build();
+client.execute(cmd);
+
+// Produces the following stack trace (truncated):
+error: query_cb err: 'Error processing incoming message: error:function_clause:[{riak_api_pb
+_server,
```
Comment (TheOnlyRew) [inline source/languages/en/riak/dev/using/2i.md]: Should this be a newline, or is it supposed to be `error:function_clause:[{riak_api_pb_server,`?

Comment (lukebakken) [inline source/languages/en/riak/dev/using/2i.md]: That's how it came out, probably due to terminal width. I'll fix it.

**Comment by lukebakken** [comment]
@TheOnlyRew this PR contains the commits from the other PR, no need to close one or the other. It'll all work out in the end.

**Comment by TheOnlyRew** [comment]
Seems strange to me that this PR hasn't updated itself now that #1618 has been merged in.
The changes look good, though, so +1.

**Pull request merged**